exploration-module/src/main/java/io/edpn/backend/exploration/adapter/persistence/MybatisBodyRepository.java

```diff
@@ -0,0 +1,123 @@
+            "ON CONFLICT (name, system_id) " +
+                    "DO UPDATE SET " +
+                    "arrival_distance = COALESCE(body.arrival_distance, EXCLUDED.arrival_distance)," +
+                    "ascending_node = COALESCE(body.ascending_node, EXCLUDED.ascending_node)," +
+                    "atmosphere = COALESCE(body.atmosphere, EXCLUDED.atmosphere)," +
+                    "atmospheric_composition = COALESCE(body.atmospheric_composition, EXCLUDED.atmospheric_composition)," +
+                    "axial_tilt = COALESCE(body.axial_tilt, EXCLUDED.axial_tilt)," +
+                    "body_composition = COALESCE(body.body_composition, EXCLUDED.body_composition)," +
+                    "discovered = COALESCE(body.discovered, EXCLUDED.discovered)," +
+                    "mapped = COALESCE(body.mapped, EXCLUDED.mapped)," +
+                    "local_id = COALESCE(body.local_id, EXCLUDED.local_id)," +
+                    "eccentricity = COALESCE(body.eccentricity, EXCLUDED.eccentricity)," +
+                    "landable = COALESCE(body.landable, EXCLUDED.landable)," +
+                    "mass = COALESCE(body.mass, EXCLUDED.mass)," +
+                    "materials = COALESCE(body.materials, EXCLUDED.materials)," +
+                    "mean_anomaly = COALESCE(body.mean_anomaly, EXCLUDED.mean_anomaly)," +
+                    "orbital_inclination = COALESCE(body.orbital_inclination, EXCLUDED.orbital_inclination)," +
+                    "parents = COALESCE(body.parents, EXCLUDED.parents)," +
+                    "periapsis = COALESCE(body.periapsis, EXCLUDED.periapsis)," +
+                    "planet_class = COALESCE(body.planet_class, EXCLUDED.planet_class)," +
+                    "radius = COALESCE(body.radius, EXCLUDED.radius)," +
+                    "rotation_period = COALESCE(body.rotation_period, EXCLUDED.rotation_period)," +
+                    "semi_major_axis = COALESCE(body.semi_major_axis, EXCLUDED.semi_major_axis)," +
+                    "surface_gravity = COALESCE(body.surface_gravity, EXCLUDED.surface_gravity)," +
+                    "surface_pressure = COALESCE(body.surface_pressure, EXCLUDED.surface_pressure)," +
+                    "surface_temperature = COALESCE(body.surface_temperature, EXCLUDED.surface_temperature)," +
+                    "system_id = COALESCE(body.system_id, EXCLUDED.system_id)," +
+                    "system_address = COALESCE(body.system_address, EXCLUDED.system_address)," +
+                    "terraform_state = COALESCE(body.terraform_state, EXCLUDED.terraform_state)," +
+                    "tidal_lock = COALESCE(body.tidal_lock, EXCLUDED.tidal_lock)," +
+                    "volcanism = COALESCE(body.volcanism, EXCLUDED.volcanism)," +
+                    "horizons = COALESCE(body.horizons, EXCLUDED.horizons)," +
+                    "odyssey = COALESCE(body.odyssey, EXCLUDED.odyssey) " +
+                    "RETURNING *"
+    })
```

Comment:
How would that look?
Im not opposed to tweaking something since its extremely bulky.
As long as this section is functional and readable though Id consider this a tangent where we could comb back over our larger queries and tweak them across the project

Reply:
most mybatis projects use an XML file where the queries are defined, this does keep theses classes a bit more clean, but sacrifices being able to easily see what the actual query is that is used.

The xml only does linking via name and afaik does not have any clickthorugh support

Reply:
Good example of a decision log item.

We went with incode text so that your query sits right above your code block. Whether you go hunt it down in XML or leave it there it's still massive and needs to be read. 

Better to be somewhere immediately visible to the code block it's for.

... Would be nice if our IDEs allowed us to auto collapse them